### PR TITLE
feature: cities

### PR DESCRIPTION
**Pull request by LukiMaurojorge**
create all back and front related to cities

view:
image

### CODE REVIEW

File: app/Http/Controllers/DeleteAirlineController.php [hunk 1]
```diff
@@ -3,14 +3,14 @@
 namespace App\Http\Controllers;
 
 use App\Models\Airline;
-use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Response;
 
 class DeleteAirlineController extends Controller
 {
-    public function __invoke(Airline $airline): JsonResponse
+    public function __invoke(Airline $airline): Response
```
Comment (gianfranco-rocco) [inline app/Http/Controllers/DeleteAirlineController.php:10]: I'm curious, why the change?

Comment (gianfranco-rocco) [inline app/Http/Controllers/DeleteAirlineController.php:10]: Why the change? Please, elaborate.

File: app/Http/Controllers/DeleteCityController.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\City;
+use Illuminate\Http\Response;
+
+class DeleteCityController extends Controller
+{
+    public function __invoke(City $city): Response
```
Comment (gianfranco-rocco) [inline app/Http/Controllers/DeleteCityController.php:10]: Pretty much the same doubt as in the `DeleteAirlineController`. Why is it responding with a `Response` instead of a `JsonResponse`?

File: database/migrations/2024_02_22_172124_create_city_table.php [hunk 3]
```diff
@@ -14,6 +14,7 @@ public function up(): void
         Schema::create('cities', function (Blueprint $table) {
             $table->id()->autoIncrement();
             $table->string('name');
+            $table->timestamps();
```
Comment (gianfranco-rocco) [inline database/migrations/2024_02_22_172124_create_city_table.php]: Remember: we shouldn't update migrations that've already been pushed. Always create a new migration. Please, create a migration for adding these to the `cities` table instead.

File: database/seeders/DatabaseSeeder.php [hunk 4]
```diff
@@ -12,5 +12,6 @@ class DatabaseSeeder extends Seeder
     public function run(): void
     {
         $this->call(AirlineSeeder::class);
+        $this->call(CitySeeder::class);
```
Comment (gianfranco-rocco) [inline database/seeders/DatabaseSeeder.php]:
```suggestion
        $this->call([
            AirlineSeeder::class,
            CitySeeder::class
        ]);
```
You can do this instead 😉

File: resources/js/city.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+            const formData = {
+                name: name,
+            };
+
+            $.ajax({
+                url: `${base_url}/api/cities/${cityId}`,
+                type: "PUT",
+                data: formData,
+            });
```
Comment (gianfranco-rocco) [inline resources/js/city.js:41]:
```suggestion
        .addEventListener("submit", function (e) {
            const cityId = document.getElementById("hidden").value;

            $.ajax({
                url: `${base_url}/api/cities/${cityId}`,
                type: "PUT",
                data: $(this).serialize(),
                contentType: 'application/json',
                dataType: 'json'
            });
```
You can do this instead (serialize a form). This prevents you from having to manually map the data into an object.

**Review by gianfranco-rocco — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/DeleteAirlineController.php (hunk 1 above)
Comment (LukiMaurojorge) [reply app/Http/Controllers/DeleteAirlineController.php:10]: thats because when I put a JsonResponse I have to transform the request to json like this 
image

File: app/Http/Controllers/DeleteCityController.php (hunk 2 above)
Comment (LukiMaurojorge) [reply app/Http/Controllers/DeleteCityController.php:10]: same

File: database/migrations/2024_02_22_172124_create_city_table.php (hunk 3 above)
Comment (LukiMaurojorge) [reply database/migrations/2024_02_22_172124_create_city_table.php]: I forgot them, I going to do that in flights table

File: resources/js/city.js (hunk 5 above)
Comment (LukiMaurojorge) [reply resources/js/city.js:41]: I think It's no possible in this case (I proved it)

File: database/migrations/2024_02_22_172124_create_city_table.php (hunk 3 above)
Comment (gianfranco-rocco) [reply database/migrations/2024_02_22_172124_create_city_table.php]: What do you mean?

File: app/Http/Controllers/DeleteAirlineController.php (hunk 1 above)
Comment (gianfranco-rocco) [reply app/Http/Controllers/DeleteAirlineController.php:10]: I guess in this case it doesn't matter because it's an empty response.

File: database/migrations/2024_04_12_190042_add_timestamp_to_city_table.php
Comment (gianfranco-rocco) [inline database/migrations/2024_04_12_190042_add_timestamp_to_city_table.php:1]: Niceee

**Review by gianfranco-rocco — APPROVED** [review]
LGTM!
